units/src/amount/signed.rs

```diff
@@ -148,16 +148,23 @@ impl SignedAmount {
     ///
     /// If the amount is too big (positive or negative) or too precise.
     pub fn from_str_in(s: &str, denom: Denomination) -> Result<SignedAmount, ParseAmountError> {
-        match parse_signed_to_satoshi(s, denom).map_err(|error| error.convert(true))? {
-            // (negative, amount)
-            (false, sat) if sat > SignedAmount::MAX.to_sat() as u64 => Err(ParseAmountError(
-                ParseAmountErrorInner::OutOfRange(OutOfRangeError::too_big(true)),
-            )),
-            (false, sat) => Ok(SignedAmount(sat as i64)), // Cast ok, value in this arm does not wrap.
-            (true, sat) if sat > SignedAmount::MIN.to_sat().unsigned_abs() => Err(
-                ParseAmountError(ParseAmountErrorInner::OutOfRange(OutOfRangeError::too_small())),
-            ),
-            (true, sat) => Ok(SignedAmount(-(sat as i64))), // Cast ok, value in this arm does not wrap.
+        let (negative, sats) =
+            parse_signed_to_satoshi(s, denom).map_err(|error| error.convert(true))?;
+
+        if negative {
+            if sats > Self::MIN.to_sat().unsigned_abs() {
+                return Err(ParseAmountError(ParseAmountErrorInner::OutOfRange(
+                    OutOfRangeError::too_small(),
+                )));
+            }
+            return Ok(Self(-(sats as i64))); // Cast ok, range checked above.
+        }
+
+        if sats > Self::MAX.to_sat() as u64 {
+            // Cast ok, Self::MAX is positive.
+            Err(ParseAmountError(ParseAmountErrorInner::OutOfRange(OutOfRangeError::too_big(true))))
+        } else {
+            Ok(Self(sats as i64)) // Cast ok, range checked above.
```

Comment:
If find the previous style much more readable. Who doesn't like it? If you insist on using `if`s just because there's a `bool` then at least using functional style (no `return` statements) would be nicer.